### PR TITLE
2023-3.1: merge deps from scipy-binder repo

### PR DESCRIPTION
**Pull request by danielballan**
I read through `environment.yml` in NSLS-II/scipy-binder line by line and made [notes](https://github.com/NSLS-II/scipy-binder/blob/notes-comparing/binder/environment.yml) on whether each item is already covered in this repo, nsls2-collection-tiled.

Note that the Python environment that the Jupyter server process runs in and the Python environment(s) of the _kernels_ that user code is executed in are distinct. This repo will only be used for kernels, not running servers, so it does not in general need to include Jupyter server extensions, for example. (Exception: Some widgets involve both server code and kernel code, and the versions between the two environments need to be compatible.)

### CODE REVIEW

**Comment by danielballan** [comment]
Note: An analogous PR will need to be made into `nsls2-collection` (sans tiled).

File: envs/env-py310.yml [hunk 1]
```diff
@@ -63,6 +70,7 @@ dependencies:
   - jupyter
   - jupyterlab
   - legacy-suitcase
+  - line_profiler
```
Comment (mrakitin) [inline envs/env-py310.yml]: We have `line_profiler` below (line 204, in the "# Profiling tools:" group).

File: envs/env-py310.yml [hunk 2]
```diff
@@ -123,10 +134,13 @@ dependencies:
   - scikit-beam >=0.0.24
   - scikit-learn
   - scipy >=1.9
+  - seaborn
+  - shadow3
```
Comment (mrakitin) [inline envs/env-py310.yml]: `shadow3` is listed below in the `# Simulation packages:` group.

File: envs/env-py310.yml [hunk 3]
```diff
@@ -123,10 +134,13 @@ dependencies:
...
   - silx
   - sixtools
   - slackclient
   - smi-analysis
+  - srwpy
```
Comment (mrakitin) [inline envs/env-py310.yml]: `srwpy` is listed below in the `# Simulation packages:` group.

**Review by mrakitin — COMMENTED** [review]
Some clean-up notes.

**Review by mrakitin — APPROVED** [review]
Looks good to me! I bumped the version of the env to `2023-3.1`.

**Comment by danielballan** [comment]
Thanks!

**Pull request merged**